src/rgw/rgw_rest_s3.cc

```diff
@@ -167,10 +167,14 @@ int RGWGetObj_ObjStore_S3::send_response_data(bufferlist& bl, off_t bl_ofs, off_
     s->cio->print("%s: %s\r\n", riter->first.c_str(), riter->second.c_str());
   }
 
-  if (!content_type)
-    content_type = "binary/octet-stream";
+  if (ret == ERR_NOT_MODIFIED) {
```

Comment:
Would you mind creating an issue at http://tracker.ceph.com/projects/rgw/issues/new for that ? It's in master and should be fixed there.